src/auxiliary/t8code.jl

```diff
@@ -254,20 +332,175 @@ function trixi_t8_fill_mesh_info(forest, elements, interfaces, mortars, boundari
+                                                                                                     1
+                            mpi_mesh_info.mpi_interfaces.local_sides[local_num_mpi_conform] = local_side
+
+                            init_mpi_interface_node_indices!(mpi_mesh_info.mpi_interfaces,
+                                                             faces, local_side, orientation,
+                                                             local_num_mpi_conform)
+
+                            neighbor_rank = remotes[findlast(ghost_remote_first_elem .<=
+                                                             neighbor_ielements[1])]
+                            mpi_mesh_info.neighbor_ranks_interface[local_num_mpi_conform] = neighbor_rank
+
+                            mpi_mesh_info.global_interface_ids[local_num_mpi_conform] = global_interface_id
+
+                            # MPI Mortar.
+                        elseif level < neighbor_level
+                            local_num_mpi_mortars += 1
+
+                            global_mortar_id = 2 * ndims(mesh) * current_linear_id + iface
+
+                            neighbor_ids = neighbor_ielements .+ 1
+
+                            local_neighbor_positions = findall(neighbor_ids .<=
+                                                               num_local_elements)
+                            local_neighbor_ids = [neighbor_ids[i]
+                                                  for i in local_neighbor_positions]
+                            local_neighbor_positions = [map_iface_to_ichild_to_position[dual_faces[1] + 1][t8_element_child_id(neighbor_scheme, neighbor_leafs[i]) + 1]
+                                                        for i in local_neighbor_positions]
+
+                            # Last entry is the large element.
+                            push!(local_neighbor_ids, current_index + 1)
+                            push!(local_neighbor_positions, 2^(ndims(mesh) - 1) + 1)
+
+                            mpi_mesh_info.mpi_mortars.local_neighbor_ids[local_num_mpi_mortars] = local_neighbor_ids
+                            mpi_mesh_info.mpi_mortars.local_neighbor_positions[local_num_mpi_mortars] = local_neighbor_positions
+
+                            init_mortar_node_indices!(mpi_mesh_info.mpi_mortars,
+                                                      (dual_faces[1], iface), orientation,
+                                                      local_num_mpi_mortars)
+
+                            neighbor_ranks = [remotes[findlast(ghost_remote_first_elem .<=
+                                                               ineighbor_ghost)]
+                                              for ineighbor_ghost in filter(x -> x >=
+                                                                                 num_local_elements,
+                                                                            neighbor_ielements)]
```

Comment:
Hm, this formatting is really unfortunate and makes the code really hard to read. Not sure if this can be improved, though.